### PR TITLE
ci setup for npm publishing #46

### LINKED ISSUE #46: Project in npm registry
Related to this issue [https://github.com/playcanvas/playcanvas-sync/issues/30](https://github.com/playcanvas/playcanvas-sync/issues/30)

We currently use this project as an npm module installed via the direct GitHub link. This causes deployment issues with CI/CD pipelines on npm v7 and later, where the deployment server requires an SSH key for public repositories. This is a known issue tracked [here](https://github.com/npm/cli/issues/2610).

I published this `playcanvas-sync` repository to the npm registry without realizing it wouldn't associate the project with the Github owners. In hindsight that does make sense. I've invited the 'playcanvas' account to become a maintainer and eventual owner of the [npm package](https://www.npmjs.com/package/playcanvas-sync), but wanted to open this line of dialogue here in case the maintainers would prefer alternative actions such as renaming the package or removing it altogether.

### PR DESCRIPTION
**Pull request by nyan-left**
Closes #46 #30

# Changelog

- Added workflow to build and publish npm package on release.
- Moved `pcsync` and `pcwatch` to standard `bin` folder.
- `bin` folder is now linted.
- Set full `eslint` path in `package.json` `lint` script.

## Remarks
The playcanvas-sync project seems to have linting setup, but there are a lot of errors.Resolving these issues and running the linter before publishing is out of scope for this PR/issue. 
The linting stage can be commented back in once the issues are resolved.  I propose the linting stage also runs on any pull requests in the future.

@willeastcott This PR will require you to set up a secret on this repo or your organisation called `NPM_TOKEN` https://docs.npmjs.com/about-access-tokens#about-granular-access-tokens

Many thanks :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,10 +41,10 @@
     "eslint": "^8.24.0"
   },
   "scripts": {
-    "lint": "eslint --ext .js,.mjs src pcsync.js pcwatch.js"
+    "lint": "./node_modules/.bin/eslint --ext .js,.mjs src bin"
```
Comment (willeastcott) [inline package.json:44]:
```suggestion
    "lint": "eslint --ext .js,.mjs src bin"
```

Comment (willeastcott) [inline package.json:44]: You shouldn't need to specify the path to `eslint`. Why is this necessary?

Comment (nyan-left) [reply package.json]: Reverted.

**Comment by nyan-left** [comment]
Hi @willeastcott,

Are there any other tweaks you'd like before we proceed with an NPM publish?

**Comment by willeastcott** [comment]
@nyan-left No, I think we're good to go. I just need to have @matthewvroman transfer the `playcanvas-sync` NPM package to me and I'll go ahead and merge this. Stay tuned. 😄

**Comment by willeastcott** [comment]
OK, the NPM package has been transferred. 🎉

**Review by willeastcott — APPROVED** [review]
